poetry/utils/env.py

```diff
@@ -448,22 +461,14 @@ def activate(self, python: str, io: IO) -> "Env":
             pass
 
         try:
-            python_version = decode(
-                subprocess.check_output(
-                    list_to_shell_command(
-                        [
-                            python,
-                            "-c",
-                            "\"import sys; print('.'.join([str(s) for s in sys.version_info[:3]]))\"",
-                        ]
-                    ),
-                    shell=True,
-                )
-            )
+            python_version = execute_python_script(
+                python,
+                "\"import sys; print('.'.join([str(s) for s in sys.version_info[:3]]))\"",
```

Comment:
What do you think about creating a constant for the script `"\"import sys; print('.'.join([str(s) for s in sys.version_info[:3]]))\""`?

Reply:
Thank you for reviewing my code. I added the constant for the script. Also, I made the `execute_python_script` a member function of `EnvManager` as it is only used inside it. It is now `EnvManager._execute_python_script`.

Reply:
Great. Good catch.

Reply:
Is there anything else I need to do? It seems that learning more test require approval.

Reply:
I think it needs more tests and one more reviewer.